### PR TITLE
Initial creation of module

### PR DESCRIPTION
**Pull request by glenjamin**
There's a whole bunch-o-stuff in this PR, I'm open to suggestions of how to slice it up to make reviewing a bit more doable.

- Extract the http instrumentation and honeycomb init code from https://github.com/geckoboard/concierge/pull/503#pullrequestreview-819090634
- setup dev and CI tooling
  - jest tests including coverage
  - typescript using JSDoc only (so no compile step is required)
  - eslint with a basic ruleset

This repo is public, with the intention that I can depend on it from private repos just using the public github URL, instead of having to mess around with private repos.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const instrumentHTTP = require('./instrumentation/http');
+
+/**
+ * @typedef {object} Options
+ * @property {string} APIKey
+ * @property {string | undefined} APIHost
+ * @property {string} TracingDataset
+ * @property {number} DesiredSampleRate
+ * @property {Record<string, unknown> | undefined} GlobalMetadata
+ */
```
Comment (BRMatt) [inline index.js:12]: Oh, can you use jsdocs to declare types without the types actually being there?

Comment (glenjamin) [reply index.js:12]: Yep, many of the typescript features can be used via JSDoc - https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html

**Review by danbahrami — COMMENTED** [review]
I think this is a nice approach Glen. Thinking about our frontend vision:
- Every JS file is a TS file

Perhaps we should think about how we would introduce a TS -> JS build step in the future

**Comment by glenjamin** [comment]
> Perhaps we should think about how we would introduce a TS -> JS build step in the future

Yeah, I think that's dependent on how we want to distribute shared modules - which I intentionally wanted to sidestep for this work.

And I think how we distribute shared modules will likely be driven by how many shared modules we want/need. Personally I favour fewer modules.

Github's got a built-in private npm thing I've never tried, but it might be good.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,5 @@
 # js-honeycomb-support
+
 Helpful stuff for adding honeycomb to a nodejs app
+
+# TODO: more docs
```
Comment (danbahrami) [inline README.md]: I think it would be helpful to have at least an example of how you would use this lib here to start with.

**Review by danbahrami — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (glenjamin) [reply README.md]: good point, I'll add that

**Pull request merged**